packages/mail/graphql/resolvers/MailLogRecord.js

```diff
@@ -0,0 +1,37 @@
+const moment = require('moment')
+
+module.exports = {
+  template: record => record.info && record.info.template,
+  subject: record => record.info && record.info.message && record.info.message.subject,
+  status: record =>
+    (record.mandrillLastEvent && record.mandrillLastEvent.msg.state) ||
+    (record.result && record.result.status) ||
+    record.status.toLowerCase(),
+  error: record =>
+    (record.mandrillLastEvent && record.mandrillLastEvent.msg && record.mandrillLastEvent.msg.diag) ||
+    (record.result && record.result.reject_reason) ||
+    (record.error && record.error.message) ||
+    (record.error && record.error.meta && record.error.meta.error && record.error.meta.error.type),
+  user: async (record, args, { loaders }) => loaders.User.byIdOrEmail.load([record.userId, record.email]),
```

Comment:
```suggestion
  user: async (record, args, { loaders }) => loaders.User.byIdOrEmail.load(record.userId || record.email),
```
would this not be enough and more deterministic. If there is a `record.userId` the email was sent to the user no matter what email the user now has, if there is no `userId` get the current user with the `email`